Source/JavaScriptCore/dfg/DFGGraph.h

```diff
@@ -1318,10 +1318,12 @@ class Graph final : public virtual Scannable {
         NodeFlags operandResultType = operand->result();
         if (operandResultType != NodeResultInt32 && immediateValue.isDouble())
             return DontSpeculateInt32;
-        
```

Comment:
Not your issue but can you change the `Node*immediate` to `Node* immediate` above?